lib/trino-parquet/src/main/java/io/trino/parquet/predicate/TupleDomainParquetPredicate.java

```diff
@@ -513,12 +513,12 @@ public static Domain getDomain(
             return Domain.all(type);
         }
 
-        long totalNullCount = nullCounts.stream()
+        Optional<Long> totalNullCount = nullCounts.map(pageNullCounts -> pageNullCounts.stream()
                 .mapToLong(value -> value)
-                .sum();
-        boolean hasNullValue = totalNullCount > 0;
+                .sum());
+        boolean hasNullValue = nullCounts.isEmpty() || totalNullCount.get() > 0;
```

Comment:
This should be moved below the next `if` block, since that no longer depends on it.